### DISCUSSION TITLE
Nextcloud AIO behind Caddy suddenly unavailable

### DISCUSSION THREAD

**Discussion by mwe120509** [nextcloud/all-in-one#4062] [Questions] [answered]
OS: ubuntu server 22.04
Nextcloud AIO v7.9.1
Community container: caddy

Nextcloud ran perfectly with that setup the last few weeks. Since today in the morning (I guess after stopping / restarting containers from automatic backups) Nextcloud is suddenly unavailable. Nothing changed in the system the last few days. (The only thing I could see is unattended security updates of ubuntu on the 12th of January.) 

2 containers seem to have issues:
nextcloud-aio-apache ("unhealthy")
nextcloud-aio-caddy ("starting" or "running" - but never "healthy" as before)

I guess, the problem is within Caddy configuration. Apache is just waiting for some Caddy config and is therefore "unhealthy".
Caddy is trying to start, but in the logs I find two lines that seem to point to errors:

```
sed: can't create temp file '/CaddyfileXXXXXX': Permission denied
...
Error: adapting config using caddyfile: parsing caddyfile tokens for 'servers': getting module named 'http.ip_sources.placeholder': module not registered: http.ip_sources.placeholder, at /Caddyfile:9
```

What I already tried and did not help:
- Restore a NC AIO backup from 13th January
- Uninstall Caddy community container and deleting all data/volumes -> still NC does not work, Apache is still waiting for some Caddy config.
- and of course Server reboot + stopping/starting the containers several times.

Hope that somebody can help. Thanks in advance.

**Comment by szaimen** [comment, marked as answer]
Hi, thanks for the report! This should be resolved with https://github.com/szaimen/aio-caddy/commit/2029e3c32d02d8e489bfdf5603bf9608c36dac14. After you add the caddy container back to the stack, it should work again.

**Reply by mwe120509** [reply]
Hi. That helped. Can confirm that it is working again. Thanks for the quick fix.
